### ISSUE TITLE
Wait for completion functions are randomly not working for few packages after 2.15.0 release

### ISSUE THREAD

**Issue by SPViradiya** [PrefectHQ/prefect#12050]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

From Friday afternoon (Sydney, Australia), suddenly DBT and Databricks job call from Prefect stopped working. It does create a sub-flow to start the task but do not create another sub-flow that will trigger a job and wait for it to finish. After digging for 3 days and trying different solutions, we figured out that there was a new version 2.15.0 released exactly around the time we started facing an issue on Friday and we always use latest version in the ECS image. We restricted ECS to use 2.14.21, the latest version before 2.15.0, everything started working fine. 

### Reproduction

```python3
Package Name: prefect-dbt
Function Name: trigger_dbt_cloud_job_run_and_wait_for_completion

Package Name: prefect-databricks
Function Name: jobs_runs_submit_and_wait_for_completion
```

### Error

```python3
There was no error at all. The job just get stuck at one point and never get terminated1
```

### Versions

```Text
2.15.0
```

**Comment by SPViradiya** [comment]
Please note that it was an issue for all flow where it has to trigger a job and wait for it.

**Comment by EmilRex** [comment]
Should be resolved by #12054

**Comment by zhen0** [comment]
@SPViradiya - I'm going to close this in favor of https://github.com/PrefectHQ/prefect/issues/12053 as we believe it's the same issue and, as Emil noted above, the immediate problem should be addressed by https://github.com/PrefectHQ/prefect/pull/12054. Please do let us know in https://github.com/PrefectHQ/prefect/issues/12053 or a new issue if you continue to see issues.
